src/vmm/src/builder.rs

```diff
@@ -576,6 +589,29 @@ fn attach_entropy_device(
     device_manager.attach_virtio_device(vm, id, entropy_device.clone(), cmdline, false)
 }
 
+fn attach_virtio_mem_device(
+    device_manager: &mut DeviceManager,
+    vm: &Arc<Vm>,
+    cmdline: &mut LoaderKernelCmdline,
+    config: &MemoryHotplugConfig,
+    event_manager: &mut EventManager,
+) -> Result<(), StartMicrovmError> {
+    let virtio_mem = Arc::new(Mutex::new(
+        VirtioMem::new(
+            vm.clone(),
```

Comment:
`Arc::clone(&vm)`

Reply:
It's a rust best practices thingy. `Arc<T>` derefs to `T`, and so by just doing `t.clone()` on a `t: &Arc<T>`, if `T` has an explicit clone function will not actually clone the `Arc`.